Controllers/PostController.cs

```diff
@@ -86,4 +86,35 @@ public IActionResult CreatePost(Post post)
         _dbContext.SaveChanges();
         return Created($"/api/post/{post.Id}", post);
     }
+
+    [HttpDelete("{postId}")]
+    [Authorize]
+    public IActionResult DeletePost( int postId)
+    {
+        Post PostToDelete = _dbContext.Posts.SingleOrDefault(p => p.Id == postId);
+        if (PostToDelete == null)
+        {
+            return NotFound();
+        }
+        else if (postId != PostToDelete.Id)
+        {
+            return BadRequest();
+        }
+        //delete comment for post
+        var relatedComments = _dbContext.Comments.Where(c => c.PostId == postId);
```

Comment:
Attempted to test if these blocks handling deleting related data are needed, but unable to confirm since we have not yet implemented the ability to post comments, reactions, or tags on posts. Let's test this commented out if we get a chance later!